src/h3lib/lib/polyfill.c

```diff
@@ -0,0 +1,593 @@
+     -2.70381656362525},
+    {-0.84291228415618, -1.26020983864103, 2.24187397694118, 1.38191906047983},
+    {-0.94271815376360, -1.32899086063916, 0.84283975752601, -0.12459257316986},
+    {-0.91898920750071, -1.32929586572429, -1.08536920415787,
+     -2.05346111080706},
+    {-0.97226652536306, -1.27950477868453, -2.58603200375485, 2.95929340513654},
+    {-1.01285145697208, -1.41845302535151, -3.13590968086981, 1.97388885726377},
+    {-1.09069387298096, -1.52480158339146, 0.28873212061306, -1.49848576331087},
+    {-1.17872424267511, -1.52480158339146, 2.53494381704943, -0.60112285060716},
+    {-1.20305471830087, -1.52480158339146, -0.60112285060716,
+     2.53494381704943}};
+
+/**
+ * For a given cell, return its bounding box. If coverChildren is true, the bbox
+ * will be guaranteed to contain its children at any finer resolution. Note that
+ * no guarantee is provided as to the level of accuracy, and the bounding box
+ * may have a significant margin of error.
+ * @param cell Cell to calculate bbox for
+ * @param out  BBox to hold output
+ * @param coverChildren Whether the bounding box should cover all children
+ */
+H3Error cellToBBox(H3Index cell, BBox *out, bool coverChildren) {
+    // Adjust the BBox to handle poles, if needed
+    int res = H3_GET_RESOLUTION(cell);
+
+    if (res == 0) {
+        int baseCell = H3_GET_BASE_CELL(cell);
+        if (NEVER(baseCell < 0) || baseCell >= NUM_BASE_CELLS) {
+            return E_CELL_INVALID;
+        }
+        *out = RES0_BBOXES[baseCell];
+    } else {
+        LatLng center;
+        H3Error centerErr = H3_EXPORT(cellToLatLng)(cell, &center);
+        if (centerErr != E_SUCCESS) {
+            return centerErr;
+        }
+        double lngRatio = 1 / cos(center.lat);
+        out->north = center.lat + MAX_EDGE_LENGTH_RADS[res];
+        out->south = center.lat - MAX_EDGE_LENGTH_RADS[res];
+        out->east = center.lng + MAX_EDGE_LENGTH_RADS[res] * lngRatio;
+        out->west = center.lng - MAX_EDGE_LENGTH_RADS[res] * lngRatio;
+    }
+
+    // Buffer the bounding box to cover children. Call this even if no buffering
+    // is required in order to normalize the bbox to lat/lng bounds
+    scaleBBox(out, coverChildren ? CHILD_SCALE_FACTOR : CELL_SCALE_FACTOR);
+
+    // Cell that contains the north pole
+    if (cell == NORTH_POLE_CELLS[res]) {
+        out->north = M_PI_2;
+    }
+
+    // Cell that contains the north pole
+    if (cell == SOUTH_POLE_CELLS[res]) {
+        out->south = -M_PI_2;
+    }
+
+    // If we contain a pole, expand the longitude to include the full domain,
+    // effectively making the bbox a circle around the pole.
+    if (out->north == M_PI_2 || out->south == -M_PI_2) {
+        out->east = M_PI;
+        out->west = -M_PI;
+    }
+
+    return E_SUCCESS;
+}
+
+/**
+ * Get a base cell by number, or H3_NULL if out of bounds
+ */
+static H3Index getBaseCell(int baseCellNum) {
+    if (NEVER(baseCellNum < 0) || baseCellNum >= NUM_BASE_CELLS) {
```

Comment:
I don't think this needs `NEVER` if the function is marked non-static and a unit test exercises this case.

Reply:
True, can update. It's not clear to me whether `static` offers any compilation advantages, or whether it just indicates a local function. We settled on `static` to avoid `_underscore` names to denote local functions, but I guess it makes sense to test this separately.